### PR TITLE
The application provides a footer for all views in which the user can navigate to the relevant sites

### PR DESCRIPTION
**Pull request by Steffengreiner**
**What was changed**

This required a major overhaul on how the main components are structured, since vaadin does NOT provide a native way to set a footer within the applayout. Therefore a lot of css and parentlayouts had to be changed to make it possible without destroying the layout. 

**NOTE** 
Data protection agreement, impressum and about us currently sports dummy content which is why this is not merged into the data manager application as of yet

### CODE REVIEW

**Comment by KochTobi** [comment]
Is the text contained correct or is it only a placeholder?

**Comment by Steffengreiner** [comment]
Since the new sites only show dummy content, this PR is stalled until we have the legal text ready

**Comment by sven1103** [comment]
@Steffengreiner there is an open merge conflict that also needs to be addressed first

**Comment by Steffengreiner** [comment]
I resolved the merge conflict, so this PR is ready when we get the text necessary for the impressum, data protection etc.

File: user-interface/src/main/resources/impressum/LegalNotice.html [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  <p>VAT identification number: DE812383453</p>
+
+  <h2 id="m169">Image credits</h2>
+  <p>Image Sources and Copyright Notices: </p>
+  <p>Mouse by Graphic Mall on IconScout</p>
+  <p>Mushroom by Jemis Mali on IconScout</p>
+  <p>Kidneys by Daniel Burka on IconScout</p>
+  <p>Liver by Daniel Burka on IconScout</p>
+  <p>Heart by Vector Stall on IconScout</p>
+  <p>Leaf by Phosphor Icons on IconScout</p>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/LegalNotice.html:30]: Please provided the actual URLs to the content and get in contact with @wow-such-code about the right attribution formatting.

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+  </p>
+
+  <h2 id="mOverview">Overview of processing operations</h2>
+  <p>The following table summarises the types of data processed, the purposes for which they are
+    processed and the concerned data subjects.</p>
+  <h3>Categories of Processed Data</h3>
+  <ul>
+    <li>Inventory data.</li>
+    <li>Employee Data.</li>
+    <li>Payment Data.</li>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/PrivacyAgreement.html:62]: remove please

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html [hunk 3]
```diff
@@ -0,0 +1,834 @@
...
+    <li>Contract data.</li>
+    <li>Usage data.</li>
+    <li>Meta, communication and process data.</li>
+    <li>Social data.</li>
+    <li>Images and/ or video recordings.</li>
+    <li>Audio recordings.</li>
+    <li>Log data.</li>
+    <li>Performance and behavioural data.</li>
+    <li>Working hours data.</li>
+    <li>Salary data.</li>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/PrivacyAgreement.html:74]: remove please

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html [hunk 4]
```diff
@@ -0,0 +1,834 @@
...
+  </ul>
+  <h3>Special Categories of Data</h3>
+  <ul>
+    <li>Health Data.</li>
+    <li>Religious or philosophical beliefs.</li>
+    <li>Trade union membership.</li>
+  </ul>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/PrivacyAgreement.html:81]: remove please

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html [hunk 5]
```diff
@@ -0,0 +1,834 @@
...
+      that the interests, fundamental rights, and freedoms of the data subject, which require the
+      protection of personal data, do not prevail.
+    </li>
+    <li><strong>Healthcare, occupational and social security processing of special categories of
+      personal data (Article 9 (2)(h) GDPR)</strong> - processing is necessary for the purposes of
+      preventive or occupational medicine, for the assessment of the working capacity of the
+      employee, medical diagnosis, the provision of health or social care or treatment or the
+      management of health or social care systems and services on the basis of Union or Member State
+      law or pursuant to contract with a health professional.
+    </li>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/PrivacyAgreement.html:135]: remove please

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html [hunk 6]
```diff
@@ -0,0 +1,834 @@
...
+  <h2 id="m15">Changes and Updates</h2>
+  <p>We kindly ask you to inform yourself regularly about the contents of our data protection
+    declaration. We will adjust the privacy policy as changes in our data processing practices make
+    this necessary. We will inform you as soon as the changes require your cooperation (e.g.
+    consent) or other individual notification.</p>
+  <p>If we provide addresses and contact information of companies and organizations in this privacy
+    policy, we ask you to note that addresses may change over time and to verify the information
+    before contacting us.</p>
+
+  <h2 id="m42">Terminology and Definitions</h2>
```
Comment (sven1103) [inline user-interface/src/main/resources/impressum/PrivacyAgreement.html:725]: Please make sure this is in sync with the data types mentioned in the introduction and contains a definition for all terms

File: user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java [hunk 7]
```diff
@@ -0,0 +1,90 @@
+package life.qbic.datamanager.views.general.footer;
```
Comment (sven1103) [inline user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java:1]: Please remove the about us page until we have content and not stalling this PR unnecessary

File: user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+            + "At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. "
+            + "Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. "
+            + "At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. "
+            + "At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet."
+            + "Duis autem vel eum iriure dolor in hendrerit in vulputate velit esse molestie consequat, vel illum dolore eu feugiat nulla facilisis at vero eros et accumsan et iusto odio dignissim qui blandit praesent luptatum zzril delenit augue duis dolore te feugait nulla facilisi. "
+            + "Lorem ipsum dolor sit amet,");
```
Comment (sven1103) [inline user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java:65]: Please remove this dummy text, this should not be in the dev branch

**Review by sven1103 — CHANGES_REQUESTED** [review]
Dear @Steffengreiner I added some comments that should be addressed.

**Comment by sven1103** [comment]
@Steffengreiner there are some merge conflicts that need to be resolved before we can merge this PR

File: user-interface/src/main/resources/impressum/LegalNotice.html (hunk 1 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/LegalNotice.html]: This is done!

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html (hunk 3 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/PrivacyAgreement.html]: Done !

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html (hunk 4 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/PrivacyAgreement.html]: Done

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html (hunk 5 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/PrivacyAgreement.html]: Done

File: user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java (hunk 7 above)
Comment (Steffengreiner) [reply user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java]: done

File: user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java (hunk 8 above)
Comment (Steffengreiner) [reply user-interface/src/main/java/life/qbic/datamanager/views/general/footer/AboutUs.java]: done

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html (hunk 2 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/PrivacyAgreement.html]: done

File: user-interface/src/main/resources/impressum/PrivacyAgreement.html (hunk 6 above)
Comment (Steffengreiner) [reply user-interface/src/main/resources/impressum/PrivacyAgreement.html:719]: done

**Review by sven1103 — APPROVED** [review]
Awesome, thank you addressing the change requests @Steffengreiner !

**Pull request merged**